packages/nodes-base/utils/utilities.ts

```diff
@@ -234,3 +235,37 @@ export function getResolvables(expression: string) {
 
 	return resolvables;
 }
+
+export const isTraversableObject = (value: any): value is JsonObject => {
```

Comment:
instead of duplicating these, I'd recommend extracting `isTraversableObject`/`removeCircularRefs` out from `packages/workflow/src/NodeErrors.ts` into `packages/workflow/src/utils.ts`.
Either that, or we need to make sure that `response.reason` is always a `NodeApiError` (which should automatically take care of circular references, instead of having to manually do this in every node).

Reply:
Ok that's what I suggested too.
In that particular case, the `response.reason` is not converted to a `NodeApiError` because of the `continueOnFail` options. It is passed directly as a node item. That's why it is not cleaned up.
I think converting `response.reason` to `NodeApiError` may leads to a lots of side effects, but we could a least "clean" the circular references at source